// r45r54r45/index.js
var token= require('jsonwebtoken');
module.exports={
    log: function(){
        console.log("it works!!");
    },
    db: {
        insert: function(tableName, data){
            var query="insert into "+tableName+" ";
            var keyList=[];
            var dataList=[];
            for(var i in data){
                keyList.push(i);
                if(typeof data[i] == "number"){
                    dataList.push(data[i]);
                }else{
                    dataList.push(escape(data[i]));
                }

            }
            query+="("+keyList.toString()+") values ("+dataList.toString()+")";
            console.log(query);
            return query;
        },
        update:function(tableName, data, whereQuery){
            var query="update "+tableName+" set ";
            var keyList=[];
            var dataList=[];
            for(var i in data){
                keyList.push(i);
                if(typeof data[i] == "number"){
                    dataList.push(data[i]);
                }else{
                    dataList.push(escape(data[i]));
                }

            }
            for(var i=0; i<keyList.length; i++){
                var key=keyList[i];
                var data=dataList[i];
                var tmpString=key+" = "+data;
                if(i!=keyList.length-1){
                    tmpString+=", ";
                }
                query+=tmpString;
            }

            query+=" where "+whereQuery;
            console.log(query);
            return query;
        },
        select: function(tableName, targetColumnArray, whereDataArray){
            var query="select ";
            var whereKeyList=[];
            var whereDataList=[];
            for(var key in whereDataArray){
                whereKeyList.push(key);
                if(typeof whereDataArray[key] == "number"){
                    whereDataList.push(whereDataArray[key]);
                }else{
                    whereDataList.push(escape(whereDataArray[key]));
                }
            }
            if(targetColumnArray.length==0){
                query+="* from "+tableName;
            }else{
                query+=targetColumnArray.toString()+" from "+tableName;
            }

            if(whereKeyList.length!=0){
                query+=" where ";
                whereDataArray.forEach(function(item, index){
                    //TODO
                });

            }
        }
    },
    token: {
        generate: function(payload, key){
            return token.sign(payload, key, {
                expiresIn: '7d'
            });
        },
        verify: function (tokenString, key) {
            return token.verify(tokenString, key);
        }
    }
}


function escape(string){
    return "'"+string+"'";
}
